src/adapters/kafka/index.ts

```diff
@@ -0,0 +1,63 @@
+import { Kafka, Consumer, Producer } from 'kafkajs'
+import Adapter from '../../lib/adapter.js'
+import GleeMessage from '../../lib/message.js'
+
+class KafkaAdapter extends Adapter {
+  private firstConnect: boolean = true
+  producer: any
+  name(): string {
+    return 'Kafka adapter'
+  }
+ 
+  async connect() {
+    const brokerUrl = new URL(this.AsyncAPIServer.url())
+    // kafka client
+    const kafka = new Kafka({
+      clientId: 'glee-app',  // clientID: hardcoded need to change afterwards 
+      brokers: [brokerUrl.host],
+      ssl: true,
+      sasl: {
+        mechanism: 'plain',
+        username: process.env.KAFKA_USERNAME,
+        password: process.env.KAFKA_PASSWORD,
+      },
+    })
+
+    const consumer = kafka.consumer({ groupId: 'glee-group' })   // groupID: hardcoded need to change afterwards
+    const producer = kafka.producer();
+    consumer.on('consumer.connect', () => {
+      if (this.firstConnect) {
+        this.firstConnect = false
+        this.emit('connect', {
+          name: this.name(),
+          adapter: this,
+          connection: consumer,
+          channels: this.channelNames
+        })
+      }
+    })
+    await consumer.connect() 
+    const subscribedChannels = this.getSubscribedChannels()
+    await consumer.subscribe({ topics: subscribedChannels, fromBeginning: true })
+    await producer.connect();
+    await consumer.run({
+      eachMessage: async ({ topic, partition, message }) => {
+        const gleemessage = new GleeMessage({
+          // topic: topic,
+          // partition: partition,
+          // value: message.value.toString(),
+          // key: message.key,
+          // offset: message.offset,
+          // timestamp: message.timestamp,
+          // headers: message.headers,
+        });
```

Comment:
Following our offline conversation. This is giving you an error because the GleeMessage constructor expects a different set of values: https://github.com/asyncapi/glee/blob/master/src/lib/message.ts#L6.

I'd group things the following way:
  * GleeMessage channel: topic.
  * GleeMessage payload: message.value. Don't call `toString()` here. Glee is capable of handling it.
  * GleeMessage headers: `partition`, `message.key`, `message.offset`, `message.timestamp`, and `message.headers`. If there's anything else in the `message` object, it should most probably be in the GleeMessage headers too.

So here's how you'd do it (beware there may be errors as I'm not testing it):

```suggestion
        const gleeMessage = new GleeMessage({
          channel: topic,
          payload: message.value,
          headers: {
            partition: partition,
            key: message.key,
            offset: message.offset,
            timestamp: message.timestamp,
            ...message.headers,
          }
        })
```

Reply:
`...message.headers`  is supposed to be like this or does it imply that there are more headers that needs to be added?

Reply:
The `message.headers` object may contain other headers that we can not know in advance. With `...message.headers` we tell JS/TS to include all of them in this object. The `...` is the [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax).